layouts/partials/js/expand.js

```diff
@@ -0,0 +1,95 @@
+// Setup all expandable tables on the page.
+var tables = document.querySelectorAll('.p-table--expanding')
+
+/**
+  Toggles the necessary aria- attributes' values on the table panels
+  to show or hide them.
+  @param {HTMLElement} element The tab that acts as the handles.
+  @param {Boolean} show Whether to show or hide the expanded row panel.
+*/
+function toggleExpanded(element, show) {
+  var target = document.getElementById(element.getAttribute('aria-controls'))
+  if (target) {
+    element.setAttribute('aria-expanded', show)
+
+    // Adjust the text of the toggle button
+    if (show) {
+      element.innerHTML = element.getAttribute('data-shown-text')
+    } else {
+      element.innerHTML = element.getAttribute('data-hidden-text')
+    }
+
+    target.setAttribute('aria-hidden', !show)
+  }
+}
+
+/**
+    Attaches event listeners for the expandable table open and close click events.
+    @param {HTMLElement} table The expandable table container element.
+  */
+function setupExpandableTable(table) {
+  // Set up an event listener on the container so that panels can be added
+  // and removed and events do not need to be managed separately.
+  table.addEventListener('click', function (event) {
+    var target = event.target
+    var isTargetOpen = target.getAttribute('aria-expanded') === 'true'
+
+    if (target.classList.contains('u-toggle')) {
+      // Toggle visibility of the target panel.
+      toggleExpanded(target, !isTargetOpen)
+    }
+  })
+}
+
+for (const table of tables) {
+  setupExpandableTable(table)
+}
+
+// Add control to expand or collapse all rows at once
+var expandButtons = document.querySelectorAll('.charmhub-toggle')
+var toggleAllButton = document.getElementById('toggle-all-rows')
+toggleAllButton.addEventListener("click", function (event) {
+    var target = event.target
+    var isTargetActive = target.getAttribute('active') === 'true'
+    // Adjust the text of the toggle button
+    target.setAttribute('active', !isTargetActive)
+    if (!isTargetActive) {
+      target.innerHTML = target.getAttribute('active-text')
+    } else {
+      target.innerHTML = target.getAttribute('inactive-text')
+    } 
+
+    for (const b of expandButtons) {
+      if (!b.disabled) {
+        toggleExpanded(b, !isTargetActive)
+      }
+    }
+  },
+  false)
+
+// Add control to only show charms in the table
+var repositoryRows = document.querySelectorAll(".repository-row")
+var toggleCharmsOnly = document.getElementById('toggle-charms-only')
```

Comment:
Please change this to a `const`

```suggestion
const toggleCharmsOnly = document.getElementById('toggle-charms-only')
```